cabal.project

```diff
@@ -9,6 +9,8 @@ packages:
 
 optimization: 0
 
+allow-newer: hedgehog-classes:hedgehog
+
```

Comment:
I intentionally did not sign-off this commit so CI would break and this would not be merged without discussing:
- Do we want to merge this with `allow-newer` (if so, I need to reword the commit message to be conventional-commits, and maybe rebase it first so both commits will build cleanly; I'll not squash, to make it trivial to revert once upstream makes a release)
- Do we want to wait for upstream to fix it (https://github.com/hedgehogqa/haskell-hedgehog-classes/pull/52)
- Did we want some other option?

Reply:
I think this is fine. It's more important that we get the GHC 9.6 work moving.

Reply:
Upstream plans to cut a new release this weekend, so I'll wait until next week to merge this.

Reply:
Let's merge. I have opened #1122 to track removing the `allow-newer` when a release happens.